File: week-02/day-01/draw_pyramid.py
# Write a program that reads a number from the standard input, then draws a
# pyramid like this:
#
#
#    *
#   ***
#  *****
# *******
#
# The pyramid should have as many lines as the number was
import re

pyramid_height = input("Enter an integer: ")
number_validator = re.compile(r"\D+")

while number_validator.match(pyramid_height) != None:
    pyramid_height = input("Enter an integer: ")

pyramid_height = int(pyramid_height)


for i in range(pyramid_height):

    print(" " * (pyramid_height-i) + "*" * (i*2) + "*")

File: week-07/day-01/draw-pyramid.rb
# // Write a program that draws a
# // pyramid like this:
# //
# //
# //    *
# //   ***
# //  *****
# // *******
# //
# // The pyramid should have as many lines as lineCount is


LINE_COUNT = 5;

LINE_COUNT.downto(0) do |space_count|
    star_count = ( LINE_COUNT - space_count ) * 2 
    puts " " * space_count + "*" * ( star_count == 0 ? star_count : star_count - 1 )
end
